server/DataAccessObjects/DriverDao.js

```diff
@@ -2,16 +2,80 @@ class DriverDao {
 
     constructor(gsDB) {
         this.gsDB = gsDB;
+        //this.driverDatabase = this.gsDB.collection("Drivers");
     }
 
     notifyAllValidDrivers(activeOrder) {
-        let driverDatabase = this.gsDB.collection("Drivers");
-        driverDatabase.where('capacity', '>=', activeOrder.getTotalQuantity()).get().then(snapshot => {
+        this.driverDatabase.where('capacity', '>=', activeOrder.getTotalQuantity()).get().then(snapshot => {
             snapshot.forEach(doc => {
                 activeOrder.notifyDriver(doc.data()["driverId"])
             });
         })
     }
+
+    createNewAccount(driver){
+        //check if driver Id doesn't exist
+        if (driver.getDriverId() == null){
```

Comment:
this should be done in when a new instance of a Driver object is created

Reply:
> this should be done in when a new instance of a Driver object is created

I put this check here so I can use generateUniqueKey() method in DriverDao.js to create new Ids. Driver.js doesn't have this method.